TopPlaces/TopPlaces/Controllers/DetailsPhotoViewController.m

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by Gennadi Iosad.
+
+#import "DetailsPhotoViewController.h"
+
+#import "PhotosHistory.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface DetailsPhotoViewController() <UIScrollViewDelegate>
+
+@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *spinner;
+
+@property (weak, nonatomic) IBOutlet UIScrollView *scrollView;
+
+@property (weak, nonatomic) IBOutlet UILabel *noImageLoadedView;
+
+@property (strong, nonatomic) UIImageView *imageView;
+
+@property (strong, nonatomic) id<CancellableTask> downloadTask;
+
+@end
+
+@implementation DetailsPhotoViewController
+
+#pragma mark -
+#pragma mark UIViewController overrides
+#pragma mark -
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  self.scrollView.delegate = self;
+  UITapGestureRecognizer *doubleTapRecognizer =
```

Comment:
We tend to extract pieces of codes like this (lines 33-38) to a method (<code>setupDoubleTapRecognizer</code> or something similar), it makes it easier to read.